### ISSUE TITLE
Unused Variables When Used As Typings

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#25]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/599

---

I'm still new to TypeScript so apologizes if I misunderstood this issue.

**What version of TypeScript are you using?**
3.2.2

**What version of `typescript-eslint-parser` are you using?**
21.0.2

**What code were you trying to parse?**

```ts
import * as fastify from 'fastify'
import { Server, IncomingMessage, ServerResponse } from 'http'

// Create a http server. We pass the relevant typings for our http version used.
// By passing types we get correctly typed access to the underlying http objects in routes.
// If using http2 we'd pass <http2.Http2Server, http2.Http2ServerRequest, http2.Http2ServerResponse>
const server: fastify.FastifyInstance<Server, IncomingMessage, ServerResponse> = fastify({})

const opts: fastify.RouteShorthandOptions = {
  schema: {
    response: {
      200: {
        type: 'object',
        properties: {
          pong: {
            type: 'string'
          }
        }
      }
    }
  }
}

server.get('/ping', opts, (request, reply) => {
  console.log(reply.res) // this is the http.ServerResponse with correct typings!
  reply.code(200).send({ pong: 'it worked!' })
})
```

From: https://github.com/fastify/fastify/blob/master/docs/TypeScript.md#example

**What did you expect to happen?**
No errors relating unused variables.

**What happened?**
It complains that `Server, IncomingMessage, ServerResponse` are all unused but technically they are used when defining the `server` variable.

**Comment by Chris911** [comment]
@JamesHenry I'm still getting this issue with v1.1.1 `@typescript-eslint/no-unused-vars` errors variable imported as types and only used for typing.

**Comment by armano2** [comment]
can you provide example of code?

**Comment by Chris911** [comment]
Here's a simple repo setup to reproduce: https://gist.github.com/Chris911/69305fa1697683b90f578e9d9244fdec

In the example `MongooseTypes` is imported and then used in another interface but eslint errors on L1.

**Comment by armano2** [comment]
yes, that's true, this is due to #21

**Comment by Chris911** [comment]
Ah ok. Thanks I'll follow that issue.
